src/signalfx_metadata.py

```diff
@@ -1228,6 +1229,22 @@ def get_collectd_version():
 
     return COLLECTD_VERSION
 
+def get_signalfx_agent_version():
+    global SIGNALFX_AGENT_VERSION
+    if SIGNALFX_AGENT_VERSION:
+        return SIGNALFX_AGENT_VERSION
+
+    SIGNALFX_AGENT_VERSION = "NOT_INSTALLED"
+    try:
+        output = popen(["/usr/bin/signalfx-agent", "-version"])
```

Comment:
Make it test for the presence of the binary first so that stand alone collectd's logs don't have needless and potentially confusing errors.